api/ombpdf/views.py

```diff
@@ -0,0 +1,83 @@
+import json
+
+from django.contrib.staticfiles.templatetags.staticfiles import static
+from django.http import FileResponse, Http404, HttpResponse
+from django.shortcuts import render
+from django.urls import reverse
+from django.utils.safestring import SafeString
+
+from ombpdf import html, rawlayout, semhtml
+from ombpdf.document import OMBDocument
+from ombpdf.download_pdfs import ROOT_DIR as DATA_DIR
+from ombpdf.util import get_ltpages
+
+
+def to_urlpath(path):
+    return '/'.join(path.relative_to(DATA_DIR).parts)
+
+
+def get_pdfmap():
+    pdfmap = {}
+    for path in DATA_DIR.glob('**/*.pdf'):
+        pdfmap[to_urlpath(path)] = path
+    return pdfmap
+
+
+_ltpages_cache = {}
```

Comment:
This is just copy-pasted now, so we can address later, but a module-level cache here feels a bit hinky. Could we use Django's caching framework instead? Then we can configure it as needed. If that seems like overkill, what do you think about the built-in [lru cache](https://docs.python.org/3/library/functools.html#functools.lru_cache)? Both prevent unbound memory leaks.

Reply:
Sure, I think lru-cache is good. Using Django's caching framework concerns me a bit b/c if we end up using an out-of-process cache, it could actually mess with my intended development use case, which is that I want the cache to be cleared whenever I make a change to the code that generates the cache's content (e.g. the core `ombpdf` python code), but *not* when I make a change to the front-end code that uses the cache's data (e.g. the HTML template and/or JS code).

Reply:
(The former will happen because of the way django's development server reboots itself whenever it detects a change to its python code--so it's convenient for the cache to have the same lifetime as the django dev server process.)